go-controller/pkg/config/config.go

```diff
@@ -406,32 +456,30 @@ func InitConfig(ctx *cli.Context, defaults *Defaults) error {
 		}
 		logrus.Infof("Parsed config file %s", f.Name())
 		logrus.Infof("Parsed config: %+v", cfg)
-
-		if defaults != nil {
-			fillDefaults(&cfg, defaults)
```

Comment:
This was one of the issues that I encountered recently. The defaults values from external_ids were fetched only if the config file was existing, otherwise it wouldn't fetch the values from external_ids. Good to see it fixed here.

Reply:
@alinbalutoiu I'd like to do testcases for the external-ids, but we need to bring in something like Kube's FakeExec first.  Will work on that in a follow-up, because I want to unit-test the CA bootstrap functionality too.